### PR TITLE
refactor(security): use controller instead of promgr in security

### PR DESCRIPTION
**Pull request by heww**
Use `project.Controller` instead of `promgr.ProjectManager` in security
implementations because we will remove `promgr` package later.

Signed-off-by: He Weiwei <hweiwei@vmware.com>

### CODE REVIEW

**Review by wy65701436 — APPROVED** [review]
lgtm

File: src/pkg/project/dao/dao.go [hunk 1]
```diff
@@ -39,6 +39,8 @@ type DAO interface {
 	GetByName(ctx context.Context, name string) (*models.Project, error)
 	// List list projects
 	List(ctx context.Context, query *q.Query) ([]*models.Project, error)
+	// Lists the roles of user for the specific project
+	ListRoles(ctx context.Context, projectID int64, userID int, groupIDs ...int) ([]int, error)
```
Comment (reasonerjt) [inline src/pkg/project/dao/dao.go:43]: It's not very clear the relationship between uid and gid?  it seems the uid MUST be a member of each GID provided?  If that's the case, would it be better if we combine the data in one data object. 

I recall models.User has the values but a lot of other irrelevant attributes.

File: src/pkg/project/manager.go [hunk 2]
```diff
@@ -46,6 +46,9 @@ type Manager interface {
 
 	// List projects according to the query
 	List(ctx context.Context, query *q.Query) ([]*models.Project, error)
+
+	// ListRoles returns the roles of user for the specific project
+	ListRoles(ctx context.Context, projectID int64, userID int, groupIDs ...int) ([]int, error)
```
Comment (reasonerjt) [inline src/pkg/project/manager.go:51]: See my comments in project/dao pkg...

File: src/pkg/project/dao/dao.go (hunk 1 above)
Comment (heww) [reply src/pkg/project/dao/dao.go:43]: It's not very well to import `pkg/user/models` here. 

We are in agreement that one package in `pkg` doesn't import other packages in the `pkg` directory.

**Review by reasonerjt — APPROVED** [review]

**Comment by reasonerjt** [comment]
I also find the way to organize project evaulator, user and builder a little confusing.  There may be a more straight forward way but it does not block merging this PR.

**Pull request merged**